duckduckgo/content-scope-scripts
# ntp: customizer drawer

**Asana Task/Github Issue:** https://app.asana.com/0/1201141132935289/1208879908085793/f

## Description

- [x] added the global setup for the customizer drawer 
- [x] everything is behind a flag, so this can be merged/released without affecting production 💪
- [x] 100% focus on performance 🚀
   - [x] uses Preact signals directly on DOM attributes to trigger the global grid animation - this prevents all VDOM diffing, making it instantaneous even on the lowest powered devices
   - [x] Rendering of children inside the drawer is deferred, to further prevent main-thread work whilst the animation is playing

## Testing Steps

- First, ensure the page operates as normal with the regular popover menu https://deploy-preview-1291--content-scope-scripts.netlify.app/build/pages/new-tab/
- Second, add `?customizerDrawer=enabled`, and verify it switches to a drawer layout, like https://deploy-preview-1291--content-scope-scripts.netlify.app/build/pages/new-tab/?customizerDrawer=enabled

**WITHOUT feature flag**
<img width="903" alt="image" src="https://github.com/user-attachments/assets/11949927-cc31-4d48-86b0-a2fbf6c43a4e">

**WITH feature flag**

NOTE: It's deliberately not including panel content, styles or anything else, this PR is designed to be small and just focus on the infrastructure changes needed.

<img width="1086" alt="image" src="https://github.com/user-attachments/assets/028ca625-e9c8-4b27-9be1-27dadfaa39cc">


## Checklist


*Please tick all that apply:*
